job_status.go

```diff
@@ -82,9 +82,6 @@ func GetJobStatus(client *buildkite.Client, org, pipeline, build, jobID string)
 	bo.MaxInterval = 10 * time.Second    // Max interval between retries
 
 	operation := func() error {
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
as above, we can still do the timeout, just passing `ctx` instead of `context.Background()` I think